sdk/servicebus/service-bus/src/serviceBusClient.ts

```diff
@@ -244,8 +245,12 @@ export class ServiceBusClient {
       | MSITokenCredentials,
     options?: ServiceBusClientOptions
   ): ServiceBusClient {
-    host = String(host);
-    const tokenProvider = new AadTokenProvider(credentials);
-    return ServiceBusClient.createFromTokenProvider(host, tokenProvider, options);
+    if (isNode) {
+      host = String(host);
+      const tokenProvider = new AadTokenProvider(credentials);
+      return ServiceBusClient.createFromTokenProvider(host, tokenProvider, options);
+    } else {
+      throw new Error("`createFromAadTokenCredentials` is not supported in browser.");
+    }
```

Comment:
Reduce nesting by
```
if (!isNode) {
    // throw error here
}
// existing code that creates and uses aad token provider

Reply:
Updated.

The reason I used `isNode` and not `!isNode` was keeping in view the rollup's "replace" plugin configuration that looks at the text " if (isNode) " and replaces it with " if (false) " in browser.
The changes seem equivalent though since we aren't enabling tree shaking. Maybe we should remove the replace plugin config? cc @bterlson 

Addressed all other comments as well, please take a look

Reply:
Yeah, better to use `if (isNode)` given the current rollup config. Short term fix might be to add the `if (!isNode)` pattern to the rollup config. Longer term fix is to have a platform.node.js and platform.browser.js that exports `isNode = true` and `isNode = false` respectively, that way either form will work fine with Rollup's tree shaker, and no replace plugin is required.

Reply:
Filed: https://github.com/Azure/azure-sdk-for-js/issues/5210